### PR TITLE
feat(rules): suspect commits and issue owner support for react-native stack frames

### PR DESCRIPTION
**Pull request by barkbarkimashark**
This PR is mostly a verification task to test react-native specific stackframes for suspect commits and issue/code owners rules evaluation. 

Doc reference: https://www.notion.so/sentry/Tech-Spec-for-Mobile-Language-Stack-Transformation-5580956cfa404a90860c3e0cbf76e2c1#8a244b5411374b0e872930c6f4320def

Fixes WOR-1804

### CODE REVIEW

File: src/sentry/tasks/groupowner.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 
 def _process_suspect_commits(
-    event_id, event_platform, event_frames, group_id, project_id, **kwargs
+    event_id, event_platform, event_frames, group_id, project_id, sdk_name=None, **kwargs
```
[the remark this answers is not in the document]
Comment (wedamija) [reply src/sentry/tasks/groupowner.py:23]: Since you've made this param optional it should be fine. Issues that can happen:
 - You add a new non-optional param. We deploy workers then webs. Webs are firing old style tasks that are missing the new parameter, so if the task runs on a new worker it fails. Avoided by making new params optional
 - You remove a parameter and don't have `**kwargs`. Workers deploy first. Now webs are firing tasks that have the old parameter, and so the task fails on new workers. Fixed by having `**kwargs`, since it eats any unexpected params.

File: src/sentry/utils/event_frames.py [hunk 2]
```diff
@@ -32,6 +34,12 @@ def cocoa_frame_munger(key: str, frame: MutableMapping[str, Any]) -> bool:
     return False
 
 
+def js_react_native_frame_munger(key: str, frame: MutableMapping[str, Any]) -> bool:
+    # intentionally left blank
+    # react-native doesn't need frame munging
+    return False
```
[the remark this answers is not in the document]
Comment (wedamija) [reply src/sentry/utils/event_frames.py:40]: I think we can remove it, just having a test for `react-native` should be enough

File: src/sentry/utils/event_frames.py [hunk 3]
```diff
@@ -1,11 +1,13 @@
 from __future__ import annotations
 
+from collections import namedtuple
 from copy import deepcopy
 from typing import Any, Callable, Mapping, MutableMapping, Optional, Sequence, Tuple, cast
 
 from sentry.utils.safe import PathSearchable, get_path
 
 FrameMunger = Callable[[str, MutableMapping[str, Any]], bool]
+SdkFrameMunger = namedtuple("SdkFrameMunger", ["frame_munger", "requires_sdk", "supported_sdks"])
```
Comment (wedamija) [inline src/sentry/utils/event_frames.py:10]: Could potentially make this a dataclass so that we can define defaults and add types for each of these. Would also help us avoid things like `munger[1]` which is a little hard to follow

File: src/sentry/tasks/groupowner.py [hunk 4]
```diff
@@ -50,7 +50,7 @@ def _process_suspect_commits(
                 "sentry.tasks.process_suspect_commits.get_serialized_event_file_committers"
             ):
                 committers = get_event_file_committers(
-                    project, group_id, event_frames, event_platform
+                    project, group_id, event_frames, event_platform, sdk_name
```
Comment (wedamija) [inline src/sentry/tasks/groupowner.py:53]: You need to pass this as a kwarg (might be safer to pass as a kwarg in other places that you pass it too, but not causing bugs currently)

Comment (barkbarkimashark) [reply src/sentry/tasks/groupowner.py:53]: Why does this need to be passed as a kwarg? The `sdk_name` is optional. 

I'm fine with adding it, just don't fully understand why its needed.

File: src/sentry/utils/event_frames.py (hunk 3 above)
Comment (barkbarkimashark) [reply src/sentry/utils/event_frames.py:10]: Good idea!

File: src/sentry/tasks/groupowner.py (hunk 4 above)
Comment (wedamija) [reply src/sentry/tasks/groupowner.py:53]: It's what's causing your test failures, since currently you're passing it to the `frame_limit` arg.

Comment (barkbarkimashark) [reply src/sentry/tasks/groupowner.py:53]: Ahh, doh...

**Review by wedamija — APPROVED** [review]
